website/docs/r/datahub_topic.html.markdown

```diff
@@ -0,0 +1,79 @@
+---
+layout: "alicloud"
+page_title: "Alicloud: alicloud_datahub_topic"
+sidebar_current: "docs-alicloud-resource-datahub-topic"
+description: |-
+  Provides a Alicloud datahub topic resource.
+---
+
+# alicloud\_datahub\_topic
+
+The topic is the basic unit of Datahub data source and is used to define one kind of data or stream. It contains a set of subscriptions. You can manage the datahub source of an application by using topics. [Refer to details](https://help.aliyun.com/document_detail/47440.html).
+
+## Example Usage
+
+Basic Usage
+
+- BLob Topic
+
+  ```
+  resource "alicloud_datahub_topic" "example" {
+    name = "tf_datahub_topic"
+    project_name = "tf_datahub_project"
+    record_type = "BLOB"
+    shard_count = 3
+    life_cycle = 7
+    comment = "created by terraform"
+  }
+  ```
+- Tuple Topic
+
+  ```
+  resource "alicloud_datahub_topic" "example" {
+    name = "tf_datahub_topic"
+    project_name = "tf_datahub_project"
+    record_type = "TUPLE"
+    record_schema = {
+      bigint_field = "BIGINT"
+      timestamp_field = "TIMESTAMP"
+      string_field = "STRING"
+      double_field = "DOUBLE"
+      boolean_field = "BOOLEAN"
+    }
+    shard_count = 3
+    life_cycle = 7
+    comment = "created by terraform"
+  }
+  ```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `name` - (Required, ForceNew) The name of the datahub topic. Its length is limited to 1-128 and only characters such as letters, digits and '_' are allowed. It is case-insensitive.
+* `project_name` - (Required, ForceNew) The name of the datahub project that this topic belongs to. It is case-insensitive.
+* `shard_count` - (Required) The number of shards this topic contains. The permitted range of values is [1, 10].
+* `life_cycle` - (Required) How many days this topic lives. The permitted range of values is [1, 7].
```

Comment:
I think set a default to life_cycle and set it to Optional will be better. shard_count, record_type and record_schema are the same.